Automattic/jetpack
# General: Connections: Add MobileMagicLink component

#### Changes proposed in this Pull Request:

* Adds ability to send magic link from Jetpack to log user in to mobile app.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This adds new functionality.
* See p9OQAC-eq-p2. 

#### Testing instructions:

Testing the error case:

- Ensure that Jetpack site is connected to a test account that **is** an a11n account
- Go to Jetpack > Dashboard
- Click **Connect to mobile WordPress app** link
- Ensure modal pops up
- Click **Send Link** button
- Ensure that an error message occurs (this is due to you being connected to an a11n account)
- Disconnect site

Testing the success case: 
- Reconnect site to a WordPress.com test user that is not your a11n account
- Click **Connect to mobile WordPress app** link
- Ensure modal pops up
- Click **Send Link** button
- Ensure that you receive email with magic link

#### Design Changes:






Before this PR, when going to the Jetpack Dashboard you would see a plain "Account Connection" card under the heading "Connections":

<img width="1093" alt="Screen Shot 2019-06-06 at 10 09 28" src="https://user-images.githubusercontent.com/348248/59045760-4b10ef00-8846-11e9-8c63-8016cf620f41.png">

This PR adds the blue link in your "Account Connection" card that says: **Connect to mobile WordPress app**:

<img width="1073" alt="Screen Shot 2019-06-06 at 10 08 43" src="https://user-images.githubusercontent.com/348248/59045725-3b91a600-8846-11e9-8ca7-c648499c9c95.png">

This link opens up a this modal:

<img width="618" alt="Screen Shot 2019-06-06 at 10 09 00" src="https://user-images.githubusercontent.com/348248/59045953-a17e2d80-8846-11e9-8239-6bf13853fb84.png">

Clicking "Cancel" or outside the modal area will dismiss it. Clicking on "Send link" will issue a request to the Jetpack backend which in turn calls the WPCOM backend. This should result in an email being sent with a link that allows you to log in to the mobile app.

As the request is being made, the following notice is displayed in the upper right corner:

<img width="306" alt="Screen Shot 2019-06-06 at 10 12 11" src="https://user-images.githubusercontent.com/348248/59046029-cecadb80-8846-11e9-9d77-12c7e937056b.png">

Upon success, this notice is displayed:

<img width="303" alt="Screen Shot 2019-06-06 at 10 12 01" src="https://user-images.githubusercontent.com/348248/59046059-dc806100-8846-11e9-90a9-994aee745257.png">

Upon error, this notice is displayed:

<img width="341" alt="Screen Shot 2019-06-06 at 10 40 34" src="https://user-images.githubusercontent.com/348248/59046438-9ed00800-8847-11e9-8e80-c5ef8aa134e0.png">

#### Proposed changelog entry for your changes:

* Admin Dashboard: Add a way for users to get a magic link to log into the mobile app.
